pandas/_libs/index.pyx

```diff
@@ -606,25 +606,111 @@ cdef class BaseMultiIndexCodesEngine:
                        in zip(self.levels, zip(*target))]
         return self._codes_to_ints(np.array(level_codes, dtype='uint64').T)
 
-    def get_indexer(self, object target, object method=None,
-                    object limit=None):
+    def get_indexer(self, object target) -> np.ndarray:
+        """
+        Gets an indexer, i.e. set of indexes into `self`'s values for the
+        values in `target`, where -1 represents a value in `target` not existing
+        in (the cross-product of) `self.levels`
+
+        Parameters
+        ----------
+        target : list-like of keys
+            Each key is a tuple, with a label for each level of the index
+
+        Returns
+        -------
+        1-dimensional array of dtype int64 of the index
+        """
         lab_ints = self._extract_level_codes(target)
+        return self._base.get_indexer(self, lab_ints)
 
-        # All methods (exact, backfill, pad) directly map to the respective
-        # methods of the underlying (integers) index...
-        if method is not None:
-            # but underlying backfill and pad methods require index and keys
-            # to be sorted. The index already is (checked in
-            # Index._get_fill_indexer), sort (integer representations of) keys:
-            order = np.argsort(lab_ints)
-            lab_ints = lab_ints[order]
-            indexer = (getattr(self._base, f'get_{method}_indexer')
-                       (self, lab_ints, limit=limit))
-            indexer = indexer[order]
-        else:
-            indexer = self._base.get_indexer(self, lab_ints)
+    @staticmethod
+    def get_indexer_and_fill(object values, object target,
```

Comment:
don't use staticmethods

Reply:
rationale here was that it makes clear that `self` is not used; reverted anyway